pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/CreateClinicalDummyKitRoute.java

```diff
@@ -170,8 +171,9 @@ public Object handle(Request request, Response response) {
                 logger.info("found randomOncHistoryDetailId " + randomOncHistoryDetailId);
                 logger.info("found short id " + maybeParticipantByParticipantId.get().getProfile().map(ESProfile::getHruid));
                 while (tries < 10 && (oncHistoryDetail == null || StringUtils.isBlank(oncHistoryDetail.getAccessionNumber())
-                        || maybeParticipantByParticipantId.isEmpty() || !participantIsEnrolled(maybeParticipantByParticipantId)
-                        || maybeParticipantByParticipantId.get().getProfile().map(ESProfile::getHruid).isEmpty())) {
+                        || maybeParticipantByParticipantId.isEmpty()
+                        || maybeParticipantByParticipantId.get().getProfile().map(ESProfile::getHruid).isEmpty()
+                        || !participantIsEnrolled(maybeParticipantByParticipantId))) {
```

Comment:
order of this checking should have been changed to check enrollment last (when we are sure there is a short id)